flocker/node/agents/test/blockdevicefactory.py

```diff
@@ -166,6 +166,9 @@ def _openstack(**config):
     a ``CinderBlockDeviceAPI``.  They will be configured to use the region
     where the server that is running this code is running.
 
+    # FLOC-1925 Make this a public helper function
+    # Move the REGION from environment parts elsewhere.
+
```

Comment:
For minimal disruption, the region-from-environment stuff should probably continue to live in exactly this function but all the rest of the logic should be delegated to the public API that can be shared with the production code-path for config loading.